azure_monitor/src/azure_monitor/auto_collection/request_metrics.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import logging
+import threading
+import time
+from http.server import HTTPServer
+
+from opentelemetry.metrics import LabelSet, Meter
+from opentelemetry.sdk.metrics import Gauge
+
+logger = logging.getLogger(__name__)
+
+_requests_lock = threading.Lock()
+requests_map = dict()
+ORIGINAL_CONSTRUCTOR = HTTPServer.__init__
+
+
+def request_patch(func):
+    def wrapper(self=None):
+        start_time = time.time()
+        func(self)
+        end_time = time.time()
+
+        with _requests_lock:
+            # Update Count
+            count = requests_map.get("count", 0)
+            requests_map["count"] = count + 1
+            # Update duration
+            duration = requests_map.get("duration", 0)
+            requests_map["duration"] = duration + (end_time - start_time)
+
+    return wrapper
+
+
+def server_patch(*args, **kwargs):
+    if len(args) >= 3:
+        handler = args[2]
+        if handler:
+            # Patch the handler methods if they exist
+            if "do_DELETE" in dir(handler):
+                handler.do_DELETE = request_patch(handler.do_DELETE)
+            if "do_GET" in dir(handler):
+                handler.do_GET = request_patch(handler.do_GET)
+            if "do_HEAD" in dir(handler):
+                handler.do_HEAD = request_patch(handler.do_HEAD)
+            if "do_OPTIONS" in dir(handler):
+                handler.do_OPTIONS = request_patch(handler.do_OPTIONS)
+            if "do_POST" in dir(handler):
+                handler.do_POST = request_patch(handler.do_POST)
+            if "do_PUT" in dir(handler):
+                handler.do_PUT = request_patch(handler.do_PUT)
+    result = ORIGINAL_CONSTRUCTOR(*args, **kwargs)
+    return result
+
+
+class RequestMetrics:
+    def __init__(self, meter: Meter, label_set: LabelSet):
+        self._meter = meter
+        self._label_set = label_set
+        # Patch the HTTPServer handler to track request information
+        HTTPServer.__init__ = server_patch
+
+        request_duration_metric = self._meter.create_metric(
+            "\\ASP.NET Applications(??APP_W3SVC_PROC??)\\Request Execution Time",
+            "Incoming Requests Average Execution Time",
+            "milliseconds",
+            int,
+            Gauge,
+        )
+        self._request_duration_handle = request_duration_metric.get_handle(
```

Comment:
Handles are also [removed](https://github.com/open-telemetry/opentelemetry-python/issues/468). This will be quite a bit of refactor work.

Reply:
get_handle= bind it would be an easy change, I believe having OT Beta specific changes in a single PR would help us understand if we are missing something